code/game/objects/structures/tank_dispenser.dm

```diff
@@ -57,24 +57,37 @@
 	if(..())
 		return 1
 	add_fingerprint(user)
-	ui_interact(user)
+	tgui_interact(user)
 
 /obj/structure/dispenser/attack_ghost(mob/user)
-	ui_interact(user)
+	tgui_interact(user)
 
-/obj/structure/dispenser/ui_interact(mob/user, ui_key = "main", datum/nanoui/ui = null, force_open = 1, master_ui = null, datum/topic_state/state = GLOB.default_state)
-	user.set_machine(src)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/structure/dispenser/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
```

Comment:
```suggestion
/obj/structure/dispenser/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
```

Reply:
`* optional force_open bool If the UI should be re-opened instead of updated.`

`force_open` is for if you want to re-open the UI if you click the object again. In my opinion, this isnt needed, since you can just close and re-open the UI itself if you want to force refresh it, but thats just me.